### PR TITLE
Refactor and support new galaxy isa data type

### PR DESCRIPTION
**Pull request by ilveroluca**
This work set out to change the galaxy wrappers to integrate support for the new ISA data type in Galaxy.  Admittedly, I fear I got a little carried away so this PR is a bit more "invasive".

The PR includes:
  * Changes to Galaxy wrappers to support Galaxy ISA data type.
  * Renaming wrapper files and tools ids with a common `isa_` prefix (I also have the corresponding changes to the `container-galaxy-k8s-runtime`, which I will contribute should you accept the changes in this PR).
  * Implemented error checking and error reporting throughout.  Failed invocations will now return error-indicating exit codes (i.e., exit code != 0) and the Galaxy wrappers check for these to ensure that errors are reported to the user.
  * Change command line interface of `run_mtblisa.py` to the familiar "subcommands" approach. This change comes along with a refactoring of the code to fully parse command line arguments with `argparse`.
  * Change command line interface of `run_mtblisa.py` to receive output path, or alternatively print to stdout (only single-file json outputs).  The tool no longer defaults to generate a file called `out.json`.
  * New tests implemented with Python's `unittest` module.  You can run all tests with `run_tests.sh`

I also put in a new separate Dockerfile to build a the container image from Alpine which is relatively lightweight.  The full image is about 270 MB, while the current image is about 390 MB.  On the other hand, it builds numpy and other heavy dependencies from source so it takes a really long time to build.  The file is called `Dockerfile.alpine` so will not be used by default (to try it you'll need to ask Docker to use it with `-f`).  Really, it was more of an exercise while looking for ideas on how to lighten the isatools container image.

Sorry for the big monster patch.  I'm afraid it'll be hard to review, but I still hope you like it :-)

### CODE REVIEW

**Comment by djcomlab** [comment]
Thanks @ilveroluca!

Regarding the image, I've actually tried to create a ISA base image to make building quicker to pull from `isatools/isatools:3.6-slim-stretch-0.9.4`. I can take a look at the Dockerfile.alpine you created and could perhaps use that in the base instead - I personally had trouble getting it to build successfully on alpine image as pandas and numpy wouldn't compile correctly for me.

Also I had updated our containers to use `isatools==0.9.4` last week on develop, so you may need to update the Dockerfile if not pulling from this isatools base image, as I see it's set to `isatools==0.8.2`.

**Comment by ilveroluca** [comment]
Yes, you're right.  I forgot to bump the isatools version in that Dockerfile.  However, do keep in mind that the PR isn't replacing the "real" Dockerfile for the container image.  We can even delete `Dockerfile.alpine` if it generates confusion.

**Comment by djcomlab** [comment]
No confusion - what I mean is I'd like to try using your `Dockerfile.alpine` to create a `isatools/isatools:3.6-alpine-0.9.4` to replace `isatools/isatools:3.6-slim-stretch-0.9.4` if yours comes out slightly smaller.

**Comment by djcomlab** [comment]
Hi @ilveroluca can you try update to use `isatools/isatools:3.6-alpine-0.9.4`? It should be image built to bits taken from your `Dockerfile.alpine`.

**Review by djcomlab — APPROVED** [review]
lgtm

**Comment by djcomlab** [comment]
Will close when can confirm the build+test passes... at the moment there seems to be a problem in running test containers on the PhenoMeNal Jenkins.

File: galaxy/isa_get_data_files.xml [hunk 1]
```diff
@@ -1,9 +1,8 @@
 <?xml version='1.0' encoding='UTF-8'?>
-<tool id="data_files" name="data_files" version="1.0">
+<tool id="isa_get_data_files" name="Get data files" version="1.0">
```
Comment (pcm32) [inline galaxy/isa_get_data_files.xml:2]: Please make sure to bump wrapper versions, specially when changing formats of tools outputs or the way that underlying tools are invoked.

**Comment by pcm32** [comment]
Please bump Galaxy wrapper's versions whenever the wrapper has a new way of invoking the tool or formats change. Thanks!

**Comment by ilveroluca** [comment]
In this case, the tool id's have changed, which effectively makes them different tools (e.g., they won't show up in old workflows).  Do you still think we should bump the versions to 2.0?

**Pull request merged**